inspirehep/modules/workflows/views/callback.py

```diff
@@ -99,26 +115,51 @@ def webcoll_callback():
     recids field.
     """
     recids = dict(request.form).get('recids', [])
-    pending_records = current_cache.get("pending_records") or dict()
-    for rid in recids:
-        if rid in pending_records:
-            objectid = pending_records[rid]
-            workflow_object = workflow_object_class.get(objectid)
-            base_url = _get_base_url()
-            workflow_object.extra_data['url'] = join(
-                base_url, 'record', str(rid)
+    results = {}
+    for recid in recids:
+        recid = int(recid)
+        if recid in results:
+            current_app.logger.warning('Received duplicated recid: %s', recid)
+            continue
+
+        pending_records = WorkflowsPendingRecord.query.filter_by(
+            record_id=recid,
+        ).all()
+        if not pending_records:
+            current_app.logger.debug(
+                'The record %s was not found on the pending list.',
+                recid,
             )
-            workflow_object.extra_data['recid'] = rid
-            workflow_object.save()
-            db.session.commit()
-            workflow_object.continue_workflow(delayed=True)
-            del pending_records[rid]
-            current_cache.set(
-                "pending_records",
-                pending_records,
-                timeout=current_app.config["PENDING_RECORDS_CACHE_TIMEOUT"]
+            results[recid] = {
+                'status': 'failed',
+                'message': 'Recid not in pending list.',
+            }
+            continue
+
+        if not len(pending_records) > 1:
```

Comment:
Currently the workflows are not able (in any case) to handle more than one record, allowing multiple will lead to some refactoring on all the code (inspire-workflows{,-files,-ui} + inspire-next).

I don't think that in the near future we are going to have any multi-record workflows, and very hopefully, that will be after migrating to invenio3 everything, and this extra table will not be needed 🤞